docs/design-docs/reporting-sot.md

```diff
@@ -0,0 +1,538 @@
+- `prb`
+- `prb_no_sop`
+- `prb_n`
+- `mki`
+- `mki_no_sop`
+- `mki_n`
+- `ratio_met`
+- `cod_met`
+- `prd_met`
+- `prb_met`
+- `mki_met`
+- `vertical_equity_met`
+- `within_05_pct`
+- `within_10_pct`
+- `within_15_pct`
+- `within_20_pct`
+
+### Implementation
+
+- Tables should be constructed using our newly implemented dbt Python model
+  support. Preference to use PySpark for speed of construction, Pandas as
+  a fallback
+- Tables should be materialized as tables, i.e. saved as Parquet files. It may
+  be worth experimenting with file partitioning, assuming the files generated
+  are sufficiently large
+- Tests and documentation should be defined as usual, in the `schema.yml` file
+  of each dbt model
+- Table should use a new prefix `sot_` in their names
+- Tables should be constructed such that adding new columns and/or groups is
+  easy and non-breaking
+- Major changes to the tables, such as changes to ratio statistic or sales
+  validation methodologies, should be accompanied by incrementing the `method`
+  column (e.g. `v1` becomes `v2`). The prior method version should *only be
+  deprecated with the consent of all major users*. Method changes should be
+  documented in the table documentation
+
+### Testing
+
+All new tables should be accompanied by dbt data tests (and unit tests where
+appropriate). Data sets should ensure the following *at minimum*:
+
+- Geographies and groups contain their representative sets, i.e. the
+  `municipality` geography contains all county municipalities, and `class`
+  contains all CCAO class codes
+- Stages contain their representative sets, i.e. `mailed`, `certified`, and
+  `board` stages exist for each year regardless of whether or not they've
+  been finalized
+- Values fall within expected ranges
+  - Counts are always greater than or equal to 0
+  - FMVs and AVs are constrained by
+- No null, inf, or NaN values are present (where appropriate, some nulls are
+  expected e.g. not all PINs are covered by some geography types)
+- Tables are unique by primary key
+- Tests are run daily (as part of the `build-daily-dbt-models` workflow)
+
+## Example use cases
+
+The Data Department commonly receives questions like, "What is the count of
+homeowner exemptions by Chicago ward?" The new tables would let us quickly and
+confidently answer that question using the following query:
+
+```sql
+SELECT *
+FROM reporting.sot_tax_exe
+WHERE year = '2022'
+AND geography_type = 'ward'
+AND geography_year = '2023'
+AND group_type = 'no_group'
+```
+
+Another common question is, "What is the percentage change in assessed value
+for small single-family homes in district X?" This would use the following
+query:
+
+```sql
+SELECT *
+FROM reporting.sot_assessment_roll
+WHERE year = '2024'
+AND geography_type = 'cook_commissioner_district'
+AND geography_id = '2'
+AND geography_year = '2023'
+AND group_type = 'class'
+AND group_id IN ('202', '203', '204')
+AND stage = 'mailed'
+```
+
+## Open Questions
+
+### Mutability
```

Comment:
This is probably the biggest outstanding question. @ccao-jardine what do you think? Is this a requirement?

Reply:
@ccao-jardine I do want to get your input on this before we merge the design doc, as your answer here would seriously change how we build this project.

Reply:
Let's start with a mutable version and go from there!